client/src/app/components/Toolbar/Toolbar.jsx

```diff
@@ -58,91 +96,112 @@ class Toolbar extends PureComponent {
-                <img
-                  alt="Plus Icon"
-                  className="toolbar__icon-plus"
-                  src={PlusIcon}
-                />
-              </a>
-              <div
-                className={classNames('toolbar__form', {
-                  hidden: !shoppingFormVisbility
-                })}
-              >
-                <CreationForm
-                  label="Create new shopping list"
-                  onSubmit={this.handleShoppingListSubmission}
-                  type="menu"
-                />
+              <div className="toolbar__icon-wrapper z-index-high">
+                <button
+                  className="toolbar__icon-link"
+                  onClick={this.handleCohortFormVisibility}
+                  type="button"
+                >
+                  <CohortIcon />
+                  <img
+                    alt="Plus icon"
+                    className="toolbar__icon-plus"
+                    src={PlusIcon}
+                  />
+                </button>
+                <div
+                  className={classNames('toolbar__form', {
+                    hidden: !cohortFormVisibility
+                  })}
+                >
+                  <CreationForm
+                    label="Create new cohort"
+                    onSubmit={this.handleCohortSubmission}
+                    type="menu"
+                  />
+                </div>
+              </div>
+              <div className="toolbar__icon-wrapper z-index-high">
+                <button
+                  className="toolbar__icon-link"
+                  onClick={this.handleShoppingListFormVisibility}
+                  type="button"
+                >
+                  <ShoppingListIcon />
+                  <img
+                    alt="Plus Icon"
+                    className="toolbar__icon-plus"
+                    src={PlusIcon}
+                  />
+                </button>
+                <div
+                  className={classNames('toolbar__form', {
+                    hidden: !shoppingFormVisibility
+                  })}
+                >
+                  <CreationForm
+                    label="Create new shopping list"
+                    onSubmit={this.handleShoppingListSubmission}
+                    type="menu"
+                  />
+                </div>
               </div>
             </div>
-          </div>
-          <div className="toolbar__logo">
-            <AppLogo />
-          </div>
-          <div className="toolbar__right">
-            <UserBar />
+            <div className="toolbar__logo">
+              <AppLogo />
+            </div>
+            <div className="toolbar__right">
+              <UserBar
+                isMenuVisible={userBarMenuVisibility}
+                onClick={this.handleUserBarMenu}
+              />
+            </div>
           </div>
         </div>
-      </div>
+        {overlayVisibility && <Overlay type={OverlayStyleType.LIGHT} />}
```

Comment:
Clever way to easily handle closing on click outside. However, I wonder if it is user friendly: when this form is visible and user decide to click something outside the toolbar (for example: list/cohort link), he won't get the effect he wants but only closing the form...

Reply:
yes, I have noticed that also. Do you have any other solution for that? 

I have been trying with recognizing e.target on every element clicked outside the form, but there are also multiple elements like spans, inputs, and form itself which shouldn't cause form to disappear when clicked - so the IF condition excluding all of these form elements was too complicated and overloaded

Reply:
I meant that maybe it would be good that each form have own  visibility handlers. They could work independetly. And it would be easy to remove/edit them or took to another place in code. Also you could add events on keypress (escape) and clickig outside only when current form is visible (not on didMount) to avoide invoke them when there is no need.
I wonder if we could use (To avoid using overlay):
- additional option in addeventlistener: capture  -  and set it to true
- detect if clicked element is a child o form to avoid invoke closing handler,